### PR TITLE
 EZP-31542: Added PHP 7.4 support  (eZ Platform 3.0)

### PR DESCRIPTION
**Pull request by adamwojs**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [EZP-31542](https://jira.ez.no/browse/EZP-31542)
| **Type**                                   | improvement
| **BC breaks**                          | no
| **Tests pass**                          | yes
| **Doc needed**                       | no

See https://github.com/ezsystems/ezplatform-kernel/pull/37

### CODE REVIEW

File: src/lib/FieldType/DataTransformer/SingleCountryValueTransformer.php [hunk 1]
```diff
@@ -36,6 +36,10 @@ public function transform($value)
             return null;
         }
 
+        if (empty($value->countries)) {
```
Comment (ViniTou) [inline src/lib/FieldType/DataTransformer/SingleCountryValueTransformer.php:39]: Should we not also check if key exist in cases like this?

File: src/lib/Tests/FieldType/DataTransformer/SingleCountryValueTransformerTest.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            new Value(['ZW' => self::COUNTRIES_INFO['ZW']]),
+            'ZW',
+        ];
+
+        yield [
+            new Value([]),
+            null,
+        ];
+
+        yield [null, null];
```
Comment (mikadamczyk) [inline src/lib/Tests/FieldType/DataTransformer/SingleCountryValueTransformerTest.php:72]:
```suggestion
        yield 'country' => [
            new Value(['ZW' => self::COUNTRIES_INFO['ZW']]),
            'ZW',
        ];
        yield 'empty_array' => [
            new Value([]),
            null,
        ];
        yield 'null' => [null, null];
```

**Comment by adamwojs** [comment]
PR updated according to code review suggestions.

File: src/lib/FieldType/DataTransformer/SingleCountryValueTransformer.php (hunk 1 above)
Comment (adamwojs) [reply src/lib/FieldType/DataTransformer/SingleCountryValueTransformer.php:40]: I decided to throw TransformationFailedException if `Alpha2` key doesn't exist.

**Pull request merged**